backend/test/features/activity-like.feature

```diff
@@ -1,42 +1,62 @@
       | Slug           |
       | karl-heinz     |
-      | peter-lustiger |
-    And I send a POST request with the following activity to "/activitypub/users/bernd-das-brot/inbox":
+      | theodor        |
+    And I send a POST request with the following activity to "/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/karl-heinz/status/faslkasa7dasfzkjn2398hsfd",
+      "id": "http://localhost:4123/api/users/theodor/status/faslkasa7dasfzkjn2398hsfd",
       "type": "Create",
-      "actor": "http://localhost:4123/activitypub/users/karl-heinz",
+      "actor": "http://localhost:4123/users/theodor",
       "object": {
-          "id": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf",
-          "type": "Article",
-          "published": "2019-02-07T19:37:55.002Z",
-          "attributedTo": "http://localhost:4123/activitypub/users/karl-heinz",
-          "content": "Hi Max, how are you?",
-          "to": "https://www.w3.org/ns/activitystreams#Public"
+        "id": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf",
+        "type": "Article",
+        "published": "2019-02-07T19:37:55.002Z",
+        "attributedTo": "http://localhost:4123/api/users/theodor",
+        "content": "Test content for liking an object?",
+        "to": "https://www.w3.org/ns/activitystreams#Public"
       }
     }
     """
 
-  Scenario: Send a like of a person to an users inbox and make sure it's added to the likes collection
-    When I send a POST request with the following activity to "/activitypub/users/karl-heinz/inbox":
+  Scenario: Send a like of a person to a users inbox and make sure the object has been liked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/peter-lustiger/status/83J23549sda1k72fsa4567na42312455kad83",
+      "id": "http://localhost:4123/api/users/karl-heinz/status/83J23549sda1k72fsa4567na42312455kad83",
       "type": "Like",
-      "actor": "http://localhost:4123/activitypub/users/peter-lustiger",
-      "object": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf"
+      "actor": "http://localhost:4123/users/karl-heinz",
+      "object": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf"
     }
     """
     Then I expect the status code to be 200
-    And the post with id "dkasfljsdfaafg9843jknsdf" has been liked by "peter-lustiger"
+    And the post with id "dkaaadsfljsdfaaaffg9843jknsdf" has been liked by "karl-heinz"
+
+  Scenario: Send an Undo of a previous like activity to a users inbox and make sure the object has been disliked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-06T23:16:21Z" title="Tuesday, May 7th 2019, 1:16:21 am +02:00">May 7, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,42 +1,62 @@
-Feature: Like an object like an article or note
+Feature: Like an object e.g. an article or note
   As a user I want to like others posts
-  Also if I do not want to follow a previous followed user anymore,
-  I want to undo the follow.
+  Also if I decided to dislike a previous liked object,
+  I want to be able to do so.
 
   Background:
     Given our own server runs at "http://localhost:4123"
     And we have the following users in our database:
       | Slug           |
       | karl-heinz     |
-      | peter-lustiger |
-    And I send a POST request with the following activity to "/activitypub/users/bernd-das-brot/inbox":
+      | theodor        |
+    And I send a POST request with the following activity to "/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/karl-heinz/status/faslkasa7dasfzkjn2398hsfd",
+      "id": "http://localhost:4123/api/users/theodor/status/faslkasa7dasfzkjn2398hsfd",
       "type": "Create",
-      "actor": "http://localhost:4123/activitypub/users/karl-heinz",
+      "actor": "http://localhost:4123/users/theodor",
       "object": {
-          "id": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf",
-          "type": "Article",
-          "published": "2019-02-07T19:37:55.002Z",
-          "attributedTo": "http://localhost:4123/activitypub/users/karl-heinz",
-          "content": "Hi Max, how are you?",
-          "to": "https://www.w3.org/ns/activitystreams#Public"
+        "id": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf",
+        "type": "Article",
+        "published": "2019-02-07T19:37:55.002Z",
+        "attributedTo": "http://localhost:4123/api/users/theodor",
+        "content": "Test content for liking an object?",
+        "to": "https://www.w3.org/ns/activitystreams#Public"
       }
     }
     """
 
-  Scenario: Send a like of a person to an users inbox and make sure it's added to the likes collection
-    When I send a POST request with the following activity to "/activitypub/users/karl-heinz/inbox":
+  Scenario: Send a like of a person to a users inbox and make sure the object has been liked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/peter-lustiger/status/83J23549sda1k72fsa4567na42312455kad83",
+      "id": "http://localhost:4123/api/users/karl-heinz/status/83J23549sda1k72fsa4567na42312455kad83",
       "type": "Like",
-      "actor": "http://localhost:4123/activitypub/users/peter-lustiger",
-      "object": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf"
+      "actor": "http://localhost:4123/users/karl-heinz",
+      "object": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf"
     }
     """
     Then I expect the status code to be 200
-    And the post with id "dkasfljsdfaafg9843jknsdf" has been liked by "peter-lustiger"
+    And the post with id "dkaaadsfljsdfaaaffg9843jknsdf" has been liked by "karl-heinz"
+
+  Scenario: Send an Undo of a previous like activity to a users inbox and make sure the object has been disliked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
```


```suggestion
    Given the activity "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf" has been liked by "karl-heinz"
    When I send a POST request with the following activity to "/users/theodor/inbox":
```

In this scenario, you never made sure that the activity was liked. Please also use better ids instead of "dkaaadsfljsdfaaaffg9843jknsdf". You could also refactor the `Background` in this scenario.